pkg/volume/volume.go

```diff
@@ -131,10 +131,8 @@ type Deleter interface {
 
 // Attacher can attach a volume to a node.
 type Attacher interface {
-	Volume
-
 	// Attach the volume specified by the given spec to the given host
-	Attach(spec *Spec, hostName string) error
+	Attach(host VolumeHost, spec *Spec, hostName string) error
```

Comment:
volume host does not need to be passed in to Attach or Detach--the plugins are already aware of it, since it is passed in on plugin initialization:
```GO
Init(host volume.VolumeHost)
```